source/index.ts

```diff
@@ -570,9 +570,9 @@ type Assert = {
 	enumCase: <T = unknown>(value: unknown, targetEnum: T) => asserts value is T[keyof T];
 	urlInstance: (value: unknown) => asserts value is URL;
 	urlString: (value: unknown) => asserts value is string;
-	truthy: (value: unknown) => asserts value is unknown;
-	falsy: (value: unknown) => asserts value is unknown;
-	nan: (value: unknown) => asserts value is unknown;
+	truthy: <T>(value: T) => asserts value is Exclude<T, Falsy>;
```

Comment:
Why not use the same type of check as https://github.com/sindresorhus/is/blob/278e0e9696b9862381a6e3cae34f756c67ef735b/source/index.ts#L294 ?

Reply:
Can you also add a test to ensure this works correctly?

Reply:
Fixed the types.

May you share some ideas about the unittest? `test/test.ts` are checking runtime behavior, and does not contain tests about whether tsc is able to narrow down types on analyzing.

Reply:
https://github.com/sindresorhus/is/blob/278e0e9696b9862381a6e3cae34f756c67ef735b/test/test.ts#L1567-L1575

Reply:
Done!